### PR TITLE
fixed #30

### LINKED ISSUE #30: Use a function for repeated piece of code
<pre>
$(echo $link | grep -aob '"' | grep -oE '[0-9]+' | sed "<val>q;d")
</pre>
is used multiple times. Write a function which takes in argument of $link and <val> and returns desired output.

### PR DESCRIPTION
**Pull request by hasan356**

### CODE REVIEW

**Comment by pandusonu2** [comment]
Firstly, I made a new change, so your branch is behind again. Lets pull it again.
Do the following:
```
git checkout master #Goes to master branch
git pull nitDgp master #Pulls from the main branch
git checkout fix #Goes to fix branch
git rebase -i master #Rebases fix over the updated master
#Save and close the prompt that appears
git push origin master
git push -f origin fix #Pushes to your repository in github "-f" for force as SHA's have changed
```

File: downloader.sh [hunk 1]
```diff
@@ -1,4 +1,10 @@
 a="$#"
+index(){
```
Comment (pandusonu2) [inline downloader.sh:1]: The function should come first.

File: downloader.sh [hunk 2]
```diff
@@ -1,4 +1,10 @@
...
+	local link=$1
+	local x=$2
+	local start_end="$(echo $link | grep -aob '"' | grep -oE '[0-9]+' | sed "x;d")"
```
Comment (pandusonu2) [inline downloader.sh]: Ok this is a  little bit hard I think (-.-)" Here is the function you need to write:-
```
index(){
	echo "$(echo $@ | grep -aob '"' | grep -oE '[0-9]+' | sed "${!#};d")"
}
```

File: downloader.sh [hunk 3]
```diff
@@ -68,12 +74,16 @@ do
 		wget -o log.txt -O $i.html -c www.mangapanda.com/$manga/$chap/$i # Downloads the main webpage
 		grep 'src=\"http' $i.html | grep 'mangapanda' > jump.txt # Gets the list of image links
 		link=$(head -n 1 jump.txt)
-		starti="$(echo $link | grep -aob '"' | grep -oE '[0-9]+' | sed "11q;d")"
-		endi="$(echo $link | grep -aob '"' | grep -oE '[0-9]+' | sed "12q;d")"
+		starti=$(index link "11q")
```
Comment (pandusonu2) [inline downloader.sh]: You are passing the arguments wrong.
It should be $link, so the function call becomes
```starti=$(index $link "11q")```
Apply it to all other places where it applies

File: downloader.sh [hunk 4]
```diff
@@ -68,12 +74,16 @@ do
...
+		endi=$(index link "12q")
+		#starti="$(echo $link | grep -aob '"' | grep -oE '[0-9]+' | sed "11q;d")"
```
Comment (pandusonu2) [inline downloader.sh]: Don't leave stray comments in patches, please delete the line or keep it, don't comment them out.
Apply it to all other places where it applies

**Review by pandusonu2 — CHANGES_REQUESTED** [review]
Ok this was little harder than I expected. Either make the changes or let me know. I will do them myself.
Do the changes AFTER you have "rebased" the branch onto updated master.(The comment before the review)
If you make changes, you have to "amend" them. 
So when you do the 
git commit, add --amend.
The command becomes ```git commit --amend```
Sorry that it got complex (-.-)" If you think its getting too complex, leave this issue and try something else, I will take care of the issue. 
Thank you for your contribution :)

**Comment by hasan356** [comment]
no i will get it done :) i will make the changes

**Review by pandusonu2 — APPROVED** [review]

**Comment by pandusonu2** [comment]
Awesome work :D Hope you learned some things :) 
Thanks for your contribution :)
You might wanna bookmark this PR as I've explained quite a few things here. 
Feel free to check out other issues to contribute to :)

**Pull request merged**